library/ssl_cli.c

```diff
@@ -309,21 +309,21 @@ static int ssl_write_supported_elliptic_curves_ext( mbedtls_ssl_context *ssl,
     unsigned char *elliptic_curve_list = p + 6;
     size_t elliptic_curve_len = 0;
     const mbedtls_ecp_curve_info *info;
-    const mbedtls_ecp_group_id *grp_id;
+    const uint16_t *tls_id;
 
     *olen = 0;
 
     MBEDTLS_SSL_DEBUG_MSG( 3,
         ( "client hello, adding supported_elliptic_curves extension" ) );
 
-    if( ssl->conf->curve_list == NULL )
+    if( mbedtls_get_supported_groups(ssl) == NULL )
         return( MBEDTLS_ERR_SSL_BAD_CONFIG );
 
-    for( grp_id = ssl->conf->curve_list;
-         *grp_id != MBEDTLS_ECP_DP_NONE;
-         grp_id++ )
+    for( tls_id = mbedtls_get_supported_groups(ssl);
```

Comment:
Perhaps consolidate this into one loop, and remove the curve check?

Reply:
Amended

Reply:
Spacing

Reply:
Amended